### PR TITLE
Issue #23561, missing u8g_esp32_spi

### PR DESCRIPTION
**Pull request by ellensp**
### Description

Enabling MKS_MINI_12864_V3 on the controller BOARD_MKS_TINYBEE results in the error
```
.pio/build/mks_tinybee/src/src/lcd/dogm/u8g_dev_uc1701_mini12864_HAL.cpp.o:(.data.u8g_dev_uc1701_mini12864_HAL_2x_hw_spi+0x8): undefined reference to `u8g_eps_hw_spi_fn(_u8g_t*, unsigned char, unsigned char, void*)
```
Found there was missing code when BOARD_MKS_TINYBEE was added into marlin.
In particular Marlin/src/HAL/ESP32/u8g_esp32_spi.cpp

Cleaned up and reorders the include so the ifdef's actully worked.
Added #ifdef ARDUINO_ARCH_ESP32 block so it doesnt break arduino ide builds.

### Requirements

BOARD_MKS_TINYBEE
MKS_MINI_12864_V3

### Benefits

Compiles as expected

### Configurations

https://github.com/MarlinFirmware/Marlin/files/7895669/MKS_TinyBee.zip

### Related Issues
Issue https://github.com/MarlinFirmware/Marlin/issues/23561

### CODE REVIEW

File: Marlin/src/HAL/ESP32/u8g_esp32_spi.cpp [hunk 1]
```diff
@@ -0,0 +1,88 @@
+#ifdef ARDUINO_ARCH_ESP32
+
+#include "../../inc/MarlinConfigPre.h"
+#ifdef MKS_MINI_12864_V3
```
Comment (thisiskeithb) [inline Marlin/src/HAL/ESP32/u8g_esp32_spi.cpp]: This should be `FYSETC_MINI_12864_2_1` so it works with the other LCD aliases:

https://github.com/MarlinFirmware/Marlin/blob/db4172b5fa05e818f42427001398db5f13d13024/Marlin/src/inc/Conditionals_LCD.h#L34-L37

Comment (ellensp) [reply Marlin/src/HAL/ESP32/u8g_esp32_spi.cpp]: I really wish these alias where gone they don't add anything and make life more complicated. But added

**Comment by ellensp** [comment]
So changing a comment breaks building? really CI? really?

**Comment by thisiskeithb** [comment]
> So changing a comment breaks building? really CI? really?

[Those redefined pin warnings are a bit concerning](https://github.com/MarlinFirmware/Marlin/runs/4867773041?check_suite_focus=true), though I'm not sure where the board mosfet order stuff stands since there still seems to be reports of it being broken on some boards.

**Comment by ellensp** [comment]
@thisiskeithb pins_ZRIB_V52.h defines HEATER_1_PIN then includes pins_MKS_BASE_common.h  which includes pins_RAMPS which uses FET_ORDER and redefines HEATER_1_PIN
Issue is not related to this PR, is another FET_ORDER issue, which should be addressed separately.

**Comment by thisiskeithb** [comment]
> which should be addressed separately.

I understand. I wasn’t suggesting it get fixed in this PR, but we all know how patches like to get combined.

I just happened to catch it and mention it.

**Pull request merged**